lib/minidoc/finders.rb

```diff
@@ -41,6 +41,11 @@ def find_one!(query, options = {})
       find_one(query, options) or raise DocumentNotFoundError
     end
 
+    def find_one_or_initialize(attributes, options = {})
```

Comment:
WDYT about adding a test which passes in an options hash? Just to cover this part of the behavior. I guess it would be, like, a sorting parameter?